src/charm.py

```diff
@@ -116,6 +116,9 @@ def reconcile(self) -> None:
     @property
     def _pebble_layer(self) -> pebble.LayerDict:
         """Return a dictionary representing a Pebble layer."""
+        # self.state will never be None at this point, but this makes mypy happy.
+        if not self.state:  # pragma: nocover
```

Comment:
Could an assert do the trick ?

Reply:
It could, but we are trying to avoid those